main/test/ca/uwaterloo/flix/language/phase/TestParser.scala

```diff
@@ -423,4 +423,13 @@ class TestParser extends FunSuite with TestUtils {
     val result = compile(input, Options.TestWithLibNix)
     expectError[ParseError](result)
   }
+
+  test("ParseError.ForEachYield.01") {
+    val input =
+      """
+        |def f(): List[Int32] = foreach () yield 1
```

Comment:
Any reason this has to be a Parse error rather than Weeder error?

Reply:
I think its fine. But I would just remove the test then.

Reply:
Why should the test be removed? I can convert it to a Weeder error in a separate PR :)

Reply:
OK. Keep it :)